src/univers/nuget.py

```diff
@@ -164,6 +164,14 @@ def __lt__(self, other):
         # Revision is the same, so ignore it for comparison purposes.
         return self._base_semver < other._base_semver
 
+    def __hash__(self):
+        return hash(
+            (
+                self._base_semver.to_tuple(),
```

Comment:
Is this containing the build segment too? Based on the issues you found, should we also subclass the "semver" library used here? Or is there an opportunity to simplify and migrate this code to us e"semantic_version" instead?

Reply:
Yeah, this contains build value too, that's the reason we are using to_tuple() here.

We would need some workaround to migrate this to `sementic_version` 
We would also need support for case insensitive comparison of prerelease in `sementic_version` 
for example, as of now in `sementic_version` `1.0.0-BETA` != `1.0.0-beta`

Reply:
> Yeah, this contains build value too, that's the reason we are using to_tuple() here.


So make sure you actually tests this. I do not see an explicit unit test for that below.

> We would also need support for case insensitive comparison of prerelease in sementic_version
for example, as of now in sementic_version 1.0.0-BETA != 1.0.0-beta

Then this is maybe another issue

- nuget treats release as case-insensitive https://github.com/NuGet/NuGet.Client/blob/9e244ebc11baf68ea2da0b7595b11726d0da99e5/test/NuGet.Core.Tests/NuGet.Versioning.Test/VersionComparerTests.cs#L13
- node-semver  may treat these as case sensitive.  https://github.com/npm/node-semver/blob/c56a701f45653940ee8536eafe43b3e46c11d6cc/test/fixtures/comparisons.js#L28

Reply:
> So make sure you actually tests this. I do not see an explicit unit test for that below.

Sure, will add a test there too.